# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/InputQueue.cpp
# Commit: 5e275c31e9f4
# Full Hash: 5e275c31e9f42def8437d7c8a87cf4cf7aabf0c6
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2020-11-11 09:55:06
# Regressor Bug: 1674694
# File Overlap Count: 1
# Description:
#   Bug 1674694 - Change APZEventResult::mHandledByRootApzc to Maybe<APZHandledResult>. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96331
# ==============================================================================

diff -r 279eb8cf0dcd -r 5e275c31e9f4 gfx/layers/apz/src/InputQueue.cpp
--- a/gfx/layers/apz/src/InputQueue.cpp	Sat Nov 07 03:43:53 2020 +0000
+++ b/gfx/layers/apz/src/InputQueue.cpp	Wed Nov 11 00:33:20 2020 +0000
@@ -32,7 +32,7 @@
 nsEventStatus InputQueue::ReceiveInputEvent(
     const RefPtr<AsyncPanZoomController>& aTarget,
     TargetConfirmationFlags aFlags, const InputData& aEvent,
-    uint64_t* aOutInputBlockId, Maybe<bool>* aOutputHandledByRootApzc,
+    uint64_t* aOutInputBlockId, Maybe<APZHandledResult>* aOutputHandledResult,
     const Maybe<nsTArray<TouchBehaviorFlags>>& aTouchBehaviors) {
   APZThreadUtils::AssertOnControllerThread();
 
@@ -42,7 +42,7 @@
     case MULTITOUCH_INPUT: {
       const MultiTouchInput& event = aEvent.AsMultiTouchInput();
       return ReceiveTouchInput(aTarget, aFlags, event, aOutInputBlockId,
-                               aOutputHandledByRootApzc, aTouchBehaviors);
+                               aOutputHandledResult, aTouchBehaviors);
     }
 
     case SCROLLWHEEL_INPUT: {
@@ -86,7 +86,7 @@
 nsEventStatus InputQueue::ReceiveTouchInput(
     const RefPtr<AsyncPanZoomController>& aTarget,
     TargetConfirmationFlags aFlags, const MultiTouchInput& aEvent,
-    uint64_t* aOutInputBlockId, Maybe<bool>* aOutputHandledByRootApzc,
+    uint64_t* aOutInputBlockId, Maybe<APZHandledResult>* aOutputHandledResult,
     const Maybe<nsTArray<TouchBehaviorFlags>>& aTouchBehaviors) {
   TouchBlockState* block = nullptr;
   bool waitingForContentResponse = false;
@@ -178,8 +178,8 @@
       INPQ_LOG("dropping event due to block %p being in slop\n", block);
       result = nsEventStatus_eConsumeNoDefault;
     } else {
-      if (aOutputHandledByRootApzc &&
-          *aOutputHandledByRootApzc == Some(false) &&
+      if (aOutputHandledResult &&
+          *aOutputHandledResult == Some(APZHandledResult::HandledByContent) &&
           !target->IsRootContent() &&
           block->GetOverscrollHandoffChain()
               ->ScrollingDownWillMoveDynamicToolbar(target)) {
@@ -193,10 +193,12 @@
         // mDispatchToContent, we need to change it to Nothing() so that
         // GeckoView can properly wait for results from the content on the
         // main-thread.
-        INPQ_LOG("changing handledByRootApzc from Some(false) to %s\n",
-                 aFlags.mDispatchToContent ? "Nothing()" : "Some(true)");
-        *aOutputHandledByRootApzc =
-            aFlags.mDispatchToContent ? Nothing() : Some(true);
+        INPQ_LOG(
+            "changing handledByRootApzc from Some(HandledByContent) to %s\n",
+            aFlags.mDispatchToContent ? "Nothing()" : "Some(HandledByRoot)");
+        *aOutputHandledResult = aFlags.mDispatchToContent
+                                    ? Nothing()
+                                    : Some(APZHandledResult::HandledByRoot);
       }
       result = nsEventStatus_eConsumeDoDefault;
     }